Electron/src/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -45,17 +65,31 @@ function decodePacket(dataType: number, dataCount: number, data: Buffer): number
     case DataTypes.FLOAT_T:
       readBytes = data.readFloatBE.bind(data)
       break
+    case DataTypes.DOUBLE_T:
+      readBytes = data.readDoubleBE.bind(data)
+      break
+    case DataTypes.CHAR:
+      readBytes = data.readUInt8.bind(data)
+      break
```

Comment:
just out of curiosity which data types would be a double or char?

Reply:
Doubles will likely need to be used for GPS Lat/Lon for a little extra precision (it is unclear if our GPS module is precise enough for this to matter, but even a small fraction of a latitude is several feet). Characters are being added to add support for string arrays to be sent by things like the autonomy system for more robust logging to basestation. All around, these changes should make rovecomm more future proof.